graylog2-inputs/src/main/java/org/graylog2/inputs/codecs/GelfChunkAggregator.java

```diff
@@ -221,19 +224,25 @@ public boolean equals(Object o) {
 
             final ChunkEntry that = (ChunkEntry) o;
 
+            if (!id.equals(that.id)) return false;
             if (firstTimestamp != that.firstTimestamp) return false;
-            if (!chunkSlotsWritten.equals(that.chunkSlotsWritten)) return false;
-            //noinspection RedundantIfStatement
-            if (!payloadArray.equals(that.payloadArray)) return false;
+            if (chunkSlotsWritten.get() != that.chunkSlotsWritten.get()) return false;
+            if (!equalsAtomicReferenceArray(this.payloadArray, that.payloadArray)) return false;
```

Comment:
I'm actually not sure if we should include those fields in the comparison at all.

Looking at the usage of the sorted eviction set, the lookup sequence is: timestamp, messageid. The other fields aren't actually ever used for the purpose of ordering the `ChunkEntry` instances in the set.

The atomic fields are merely there to avoid a mutex around both field mutations.

Can we safely get rid of these two comparisons here and their calculation of `hashCode` and document the fact why they aren't included?